src/main/java/org/killbill/billing/client/KillBillHttpClient.java

```diff
@@ -391,87 +391,121 @@ public Response doOptions(final String uri, final Multimap<String, String> optio
             }
         }
 
+        final Response response = request(builder, timeoutSec);
+        if (response.getStatusCode() == 404 || response.getStatusCode() == 204) {
+            return createEmptyResult(clazz);
+        }
```

Comment:
I thought we said that if we add A `Location` header (and  the `followLocation= true`) we should follow even for status different than `201` ?